packages/store/addon/-private/identifiers/cache.ts

```diff
@@ -157,18 +163,31 @@ export class IdentifierCache {
       return resource;
     }
 
+    let lid = coerceId(resource.lid);
+    let identifier: StableRecordIdentifier | undefined = lid !== null ? this._cache.lids[lid] : undefined;
+
+    if (identifier !== undefined) {
+      return identifier;
+    }
+
+    const _resource = resource as ResourceIdentifierObject;
```

Comment:
We don't use const if it's not inside module scope. 
Lets try to rewrite this so TS knows that this is a resourceIdentifier

Reply:
The same way we have `isStableIdentifier` this might be more readable if we added `isIdentifier`

Reply:
For future knowledge, do we prefix scoped variables with `_`?  Based on usage in other languages, I tend to associate it with an unused variable that is given a descriptive name...